src/core/algorithms/fd/fd_algorithm.cpp

```diff
@@ -51,7 +52,7 @@ std::vector<Column const*> FDAlgorithm::GetKeys() const {
 
     for (FD const& fd : FdList()) {
         Vertical const& lhs = fd.GetLhs();
-
+        if (lhs.GetArity() > max_lhs_) continue;
```

Comment:
strange check, in the case when `max_lhs_` is treated correctly by the algorithm, `FdList()` couldn't possibly return `lhs` with arity greater than `max_lhs_`